File: temp_frontend/src/types/subgroups.ts
export interface SubGroup {
  id: number
  name: string
  sort_order?: number
  outbounds: string[]
  subscription_url?: string
  subscription_url_clash?: string
  allow_insecure?: boolean
  auto_update_last_at?: number
  auto_update_failed_sources?: string | string[]
  auto_update_error?: string
  createdAt?: string
  updatedAt?: string
}

export function createSubGroup(data?: Partial<SubGroup>): SubGroup {
  return {
    id: 0,
    name: '',
    sort_order: 0,
    outbounds: [],
    subscription_url: '',
    subscription_url_clash: '',
    allow_insecure: false,
    auto_update_last_at: 0,
    auto_update_failed_sources: '',
    auto_update_error: '',
    ...data
  }
}

File: database/model/subgroups.go
package model

import (
	"time"
)

// SubGroup stores subscription-manager groups.
type SubGroup struct {
	Id                      uint      `json:"id" gorm:"primaryKey;autoIncrement"`
	Name                    string    `json:"name" gorm:"unique;not null"`
	SortOrder               int       `json:"sort_order" gorm:"default:0;index"`
	Outbounds               string    `json:"outbounds" gorm:"type:text"`
	SubscriptionUrl         string    `json:"subscription_url" gorm:"type:text"`
	SubscriptionUrlClash    string    `json:"subscription_url_clash" gorm:"type:text"`
	AllowInsecure           bool      `json:"allow_insecure" gorm:"default:false"`
	AutoUpdateLastAt        int64     `json:"auto_update_last_at" gorm:"default:0"`
	AutoUpdateFailedSources string    `json:"auto_update_failed_sources" gorm:"type:text"`
	AutoUpdateError         string    `json:"auto_update_error" gorm:"type:text"`
	CreatedAt               time.Time `json:"created_at"`
	UpdatedAt               time.Time `json:"updated_at"`
}

func (SubGroup) TableName() string {
	return "sub_groups"
}
